flink-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/proxy/DynamoDBStreamsProxy.java

```diff
@@ -110,6 +111,24 @@ public GetShardListResult getShardList(Map<String, String> streamNamesWithLastSe
         return result;
     }
 
+    @Override
+    public String getShardIterator(
+            StreamShardHandle shard, String shardIteratorType, @Nullable Object startingMarker)
+            throws InterruptedException {
+        try {
+            return super.getShardIterator(shard, shardIteratorType, startingMarker);
+        } catch (ResourceNotFoundException re) {
+            if (LOG.isInfoEnabled()) {
```

Comment:
nit: I do not think this adds any value, since the check is performed under the hood.

Reply:
I agree, we risk polluting the logs for no value. Can have a comment here instead to note that sometimes the API provides invalid shards?
```
// describeStream also returns shards that are no longer valid which need to be marked as complete.
```

Reply:
Given the expected frequency of this (edge case), we should leave the log here.

Reply:
Sorry, I didn't realise you were referring to the logging check. Thanks for confirming that it's performed under the hood, I'm removing it (and leaving the log).